core/test/utilities/helpers/routing-helpers.spec.js

```diff
@@ -348,6 +348,28 @@ describe('Routing-helpers', () => {
     });
   });
 
+  describe('getContext', () => {
+    const context = { someValue: 'foo' };
+    const node = {
+      context,
+      externalLink: { url: 'https://someurl.com', sameWindow: false },
+      label: "External link"
+    };
+    it('get context from node', () => {
+      assert.deepEqual(RoutingHelpers.getContext(node, undefined), context);
+    });
+    it('get context directly', () => {
+      const ctx = { someValue: 'bar' };
+      assert.deepEqual(RoutingHelpers.getContext(node, ctx), ctx);
+    });
+    it('get context with parent node', () => {
+      const parentContext = { otherValue: 'parent' };
+      const nodeWithParent = { ...node, parent: { context: parentContext } };
+      const expect = { ...context, ...parentContext };
+      assert.deepEqual(RoutingHelpers.getContext(nodeWithParent, undefined), expect);
```

Comment:
(n2h) Here as well, it costs us nothing but we'll be sure it works as expected :).